src/ray/util/pipe_logger.cc

```diff
@@ -33,119 +33,50 @@ namespace ray {
 
 namespace {
 
-// Default pipe log read buffer size.
-constexpr size_t kDefaultPipeLogReadBufSize = 1024;
-
-size_t GetPipeLogReadSizeOrDefault() {
-  // TODO(hjiang): Write a util function `GetEnvOrDefault`.
-  const char *var_value = std::getenv(kPipeLogReadBufSizeEnv.data());
-  if (var_value != nullptr) {
-    size_t read_buf_size = 0;
-    if (absl::SimpleAtoi(var_value, &read_buf_size) && read_buf_size > 0) {
-      return read_buf_size;
-    }
-  }
-  return kDefaultPipeLogReadBufSize;
-}
-
 struct StreamDumper {
   absl::Mutex mu;
   bool stopped ABSL_GUARDED_BY(mu) = false;
   std::deque<std::string> content ABSL_GUARDED_BY(mu);
 };
 
-// File descriptors which indicates standard stream.
-#if defined(__APPLE__) || defined(__linux__)
-struct StdStreamFd {
-  int stdout_fd = STDOUT_FILENO;
-  int stderr_fd = STDERR_FILENO;
-};
-#elif defined(_WIN32)
-// TODO(hjiang): not used for windows, implement later.
-struct StdStreamFd {
-  int stdout_fd = -1;
-  int stderr_fd = -1;
+// Used to write to dup-ed stdout and stderr; use shared pointer to make it copy
+// constructible.
+struct StreamSink {
+  std::shared_ptr<boost::iostreams::stream<boost::iostreams::file_descriptor_sink>>
+      stdout_sink;
```

Comment:
```suggestion
      stdout_ostream;
```

we are storing `boost::iostreams::stream` not `file_descriptor_sink`. StreamSink needs to be renamed to reflect it as well (maybe `StdOstream`)